src/main/scala/com/wavesplatform/metrics/Metrics.scala

```diff
@@ -25,6 +28,11 @@ object Metrics extends ScorexLogging {
 
   case class Settings(enable: Boolean, nodeId: Int, influxDb: InfluxDbSettings)
 
+  object Settings {
+    import com.wavesplatform.settings._ // implicit hypherCase
+    implicit val metricsSettingsValueReader: ValueReader[Settings] = arbitraryTypeValueReader
```

Comment:
Please remove all explicit usages of `arbitraryTypeValueReader`, just one import in `WavesSettings.scala` seems to do the trick.

Reply:
But then a macro reader takes priority and ignores all custom writers from companion objects.